src/Dotenv.php

```diff
@@ -35,8 +36,6 @@ public function __construct($path, $file = '.env')
      */
     public function load()
     {
-        $this->loader = new Loader($this->filePath, $immutable = true);
```

Comment:
I don't think this can be removed, but there is not a test case to cover the scenario for why.

Basically, if you call `overload`, and then `load` on a different file, the same loader will be retained from `overload`, which will leave the `immutable` variable set to `false` (from the `overload` method), and the expected behavior will diverge from what this method is supposed to do. A test case needs to be added to ensure this does not happen, and this line probably needs to be added back in. @GrahamCampbell